dagshub/data_engine/model/query_result.py

```diff
@@ -482,6 +489,81 @@ def dp_url(dp: Datapoint):
+                annotations.extend(dp.metadata[annotation_field].annotations)
+        return annotations
+
+    def export_as_yolo(
+        self,
+        download_dir: Optional[Union[str, Path]] = None,
+        annotation_field: Optional[str] = None,
+        annotation_type: Optional[Literal["bbox", "segmentation", "pose"]] = None,
+    ) -> Path:
+        """
+        Downloads the files and annotations in a way that can be used to train with YOLO immediately.
+
+        Args:
+            download_dir: Where to download the files. Defaults to ``./dagshub_export``
+            annotation_field: Field with the annotations. If None, uses the first alphabetical annotation field.
+            annotation_type: Type of YOLO annotations to export.
+                Possible values: "bbox", "segmentation", "pose".
+                If None, returns based on the most common annotation type.
+
+        Returns:
+            The path to the YAML file with the metadata. Pass this path to ``YOLO.train()`` to train a model.
+        """
+        if annotation_field is None:
+            annotation_field = sorted([f.name for f in self.fields if f.is_annotation()])[0]
+            log_message(f"Using annotations from field {annotation_field}")
+
+        if download_dir is None:
+            download_dir = Path("dagshub_export")
+        download_dir = Path(download_dir) / "data"
+
+        annotations = self._get_all_annotations(annotation_field)
+
+        categories = Categories()
+        ann_types: dict[type, int] = {}
+
+        for ann in annotations:
+            if annotation_type is None:
+                ann_type = type(ann)
+                ann_types[ann_type] = ann_types.get(ann_type, 0) + 1
+            for cat in ann.categories.keys():
+                categories.get_or_create(cat)
+
+        if annotation_type is None:
+            annotation_types = [type(ann) for ann in annotations]
+            counter = Counter(annotation_types)
+            most_common = counter.most_common(1)
+            annotation_type = ir_mapping[most_common[0][0]]
+            log_message(f"Importing annotations of type {annotation_type} (most common type of annotation).")
+
+        image_download_path = Path(download_dir)
+
+        # Add the source prefix to all annotations
+        for ann in annotations:
+            ann.filename = os.path.join(self.datasource.source.source_prefix, ann.filename)
+
+        # If there's no folder "images" in the datasource, prepend it to the path
+        if not any("images/" in ann.filename for ann in annotations):
+            for ann in annotations:
+                ann.filename = os.path.join("images", ann.filename)
+            image_download_path = Path(download_dir) / "images"
+
+        context = YoloContext(annotation_type=annotation_type, categories=categories)
+        context.path = image_download_path
+
+        log_message("Downloading image files...")
+        self.download_files(image_download_path)
```

Comment:
Can I not download the actual images if I have them?

Reply:
If you have them at the exact path they will be downloaded to, then the downloader will skip them.
I don't expect users to guess exactly where it will get downloaded to though to be honest